helm/values.yaml

```diff
@@ -44,32 +58,71 @@ service:
     port: 80
     targetPort: 8000
     annotations:
-      external-dns.alpha.kubernetes.io/hostname: substrate-telemetry.parity-stg.parity.io.
+      external-dns.alpha.kubernetes.io/hostname: submit.telemetry.parity-stg.parity.io.
   core:
     type: LoadBalancer
     port: 80
     targetPort: 8000
     annotations:
       cloud.google.com/load-balancer-type: Internal
       networking.gke.io/internal-load-balancer-allow-global-access: "true"
-      external-dns.alpha.kubernetes.io/hostname: substrate-telemetry-core.parity-stg.parity.io.
-
+      external-dns.alpha.kubernetes.io/hostname: feed.telemetry.parity-stg.parity.io.
+  frontend:
+    type: ClusterIP
+    port: 80
+    targetPort: 8000
+    annotations: {}
 
 ingress:
-  enabled: false
-  className: ""
-  annotations: {}
-    # kubernetes.io/ingress.class: nginx
-    # kubernetes.io/tls-acme: "true"
-  hosts:
-    - host: chart-example.local
-      paths:
-        - path: /
-          pathType: ImplementationSpecific
-  tls: []
-  #  - secretName: chart-example-tls
-  #    hosts:
-  #      - chart-example.local
+  shard:
```

Comment:
It's hard for me to follow this too much without just deploying it and seeing what it looks like when deployed.

Does ingress being `enabled: false` for the shard and core mean that neither are exposed from k8s (the shard's `/submit` endpoint and core's `/feed` endpoints need to be reachable outside)?

I didn't notice anything in this PR but might have missed it: did you have any thoughts on how to *only* expose the shard's `/submit` endpoint and core's `/feed` endpoint outside of k8s and keep the other paths internal-only?

Ultimately, for me, it's probably easiest if we try deploying whatever we have and then seeing whether it works and looks sensible from there. I'm sure there will be various bits we'll want to tweak when we see how it all works!

Reply:
there are two ways for exposing a service to outside of the Kubernetes cluster:
1- using ingress
2- using a dedicated load balancer

I used a dedicated load balancer for shards and core service, because I didn't want to pass the traffic through the ingress controller as it might affect other deployments in the cluster, so I decided to have a separate loadbalancer for the telemetry components for now. The disadvantage of using loadbalancers is that they don't understand layer 7 traffic as they only operate on layer 4(TCP/UDP).
for production, I would suggest creating a dedicated ingress controller for telemetry service and expose all the services using that ingress controller. this way we would be able to have layer 7 decisions like blocking or allowing traffic to certain endpoints and because this is a dedicated ingress controller, it wouldn't affect other deployments.